x/lockup/keeper/lock.go

```diff
@@ -48,18 +48,22 @@ func (k Keeper) BeginUnlockAllNotUnlockings(ctx sdk.Context, account sdk.AccAddr
 }
 
 // AddToExistingLock adds the given coin to the existing lock with the same owner and duration.
-// Returns an empty array of period lock when a lock with the given condition does not exist.
-func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) ([]types.PeriodLock, error) {
+// Returns the updated lock ID if successfully added coin, returns 0 when a lock with
+// given condition does not exist.
+func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sdk.Coin, duration time.Duration) (uint64, error) {
 	locks := k.GetAccountLockedDurationNotUnlockingOnly(ctx, owner, coin.Denom, duration)
 	// if existing lock with same duration and denom exists, just add there
 	if len(locks) > 0 {
+		// there should only be a single lock with the same duration + token, thus we take the first lock
 		lock := locks[0]
 		_, err := k.AddTokensToLockByID(ctx, lock.ID, owner, coin)
 		if err != nil {
-			return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
+			return 0, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, err.Error())
 		}
+		return lock.ID, nil
 	}
-	return locks, nil
+
+	return 0, nil
```

Comment:
Since there is no existing lock, why wouldn't this be an error?

Reply:
it's mainly because of how this API is getting used. 

this API is getting called in the msg server, we attempt to add to existing lock before creating a new lock with the user + duration combination.

Theefore, in this case, it's not an error that we couldn't find a lock with the user + duration combination.

A idea that pops up in my mind rn is maybe changing this API to return a boolean value as well to indicate whether it has been added to existing lock or not?

lmk what you think

Reply:
How do you think the boolean would help? I might not be seeing it but I'm afraid that it would only overcomplicate the logic of the caller.

In my opinion, independent of the solution chosen, we should decouple the implementation details of this method from its caller.

I'd like to suggest returning an error here and then let the caller decide how they want to handle it. That is, the caller can check what error it is specifically and continue normally if the error is deemed acceptable.

Another idea is to implement a method `HasLock()` that we call before calling this method. At the same time, we can make this method assume that the lock already exists. If not, an error would be an expected return here.

Let me know what you decide

Reply:
The original intention behind my suggestion of adding a boolean field was to separate real errors and simple cases without matching locks, but I think your solution works as well!

Just added `HasLock` check!

Reply:
Awesome! So do you think it makes sense to return an error now here, making this method only be successful if the lock exists?

Reply:
Makes sense, this change makes it state breaking though right?

Reply:
Yes, it does. We shouldn't backport this

Reply:
hrrmmm do you think this change is worth doing if its state breaking?